speech/google/cloud/speech/sample.py

```diff
@@ -95,10 +99,15 @@ def source_uri(self):
     def content(self):
         """Bytes of audio content.
 
-        :rtype: bytes
-        :returns: Byte stream of audio content.
+        :rtype: bytes or file
+        :returns: Byte stream of audio content or file like object.
         """
-        return self._content
+        if self._content is None:
+            return self._content
+        content = self._content
+        if getattr(self._content, 'read', None) is not None:
+            content = self._content.read()
```

Comment:
This method will consume the "entire" stream without going back to the beginning. That's not always bad, but here, you don't make any "note" of that on the object, so callers have no real way of knowing if they are getting the whole thing.

We need a better strategy here. A "simple" answer might be to just over-write the file-like object with the consumed content returned by `.read()`, though that doesn't fit your target examples of a stream from a microphone (since there is no such thing as "all" or "the end").